### PR TITLE
Authentication, article CR, comment R

### PR DESCRIPTION
**Pull request by Apocalam**
No CSS yet

### CODE REVIEW

File: src/app/app-routing.module.spec.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { AppRoutingModule } from './app-routing.module';
```
Comment (jpbalarini) [inline src/app/app-routing.module.spec.ts]: please remove files that are autogenerated and do not have any functionality

File: src/app/articles/articles.component.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import { Comment } from '../models/comment';
+import { User } from '../models/user';
+
+@Component({
+  selector: 'app-articles',
+  templateUrl: './articles.component.html',
+  styleUrls: ['./articles.component.css']
+})
+export class ArticlesComponent implements OnInit {
+
```
Comment (jpbalarini) [inline src/app/articles/articles.component.ts]: please remove blank line

File: src/app/articles/articles.component.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  private url = environment.APIUrl + '/articles';
+  articles: Article[] = [];
+  article: Article;
+  comments: Comment[] = [];
+  currentUser: User;
+
+  constructor(private articleService: ArticleService,
+              private commentsService: CommentsService,
+              private userService: UserService) {
+    }
```
Comment (jpbalarini) [inline src/app/articles/articles.component.ts]: check indentation, is this ok?

File: src/app/articles/articles.component.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+  getComments(id: number): void {
+    this.commentsService.getComments(id).subscribe(
+      data => this.comments = data
+    );
+  }
+
+  newArticle(title: string, text: string): void {
+    title = title.trim();
+    if (!title) { return; }
```
Comment (jpbalarini) [inline src/app/articles/articles.component.ts]: inline if FTW

File: src/app/comments/comments.component.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import { Comment } from '../models/comment';
+import { environment } from '../../environments/environment';
+
+@Component({
+  selector: 'app-comments',
+  templateUrl: './comments.component.html',
+  styleUrls: ['./comments.component.css']
+})
+export class CommentsComponent implements OnInit {
+
```
Comment (jpbalarini) [inline src/app/comments/comments.component.ts]: please remove this line

File: src/app/login/login.component.css [hunk 6]
```diff
@@ -0,0 +1,43 @@
+.container {
+  display: flex;
```
Comment (jpbalarini) [inline src/app/login/login.component.css]: 👍 flex FTW

File: src/app/login/login.component.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+import { User } from '../models/user';
+
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+
```
Comment (jpbalarini) [inline src/app/login/login.component.ts]: please remove this line

File: src/app/login/login.component.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    this.authenticationService.login(this.user).subscribe(
+      data => {
+        this.router.navigate([this.returnUrl]);
+      },
+      error => console.log("error")
+    );
+  }
+
+  get diagnostic() { return JSON.stringify(this.user) }
+
```
Comment (jpbalarini) [inline src/app/login/login.component.ts]: unnecessary line

File: src/app/models/user.ts [hunk 9]
```diff
@@ -0,0 +1,14 @@
+export class User {
+
```
Comment (jpbalarini) [inline src/app/models/user.ts]: unnecessary line

File: src/app/page-not-found/page-not-found.component.ts [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+  templateUrl: './page-not-found.component.html',
+  styleUrls: ['./page-not-found.component.css']
+})
+export class PageNotFoundComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit() {
+  }
+
```
Comment (jpbalarini) [inline src/app/page-not-found/page-not-found.component.ts]: remove this line

File: src/app/services/articles.service.ts [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+    );
+  }
+
+  newArticle(article: Article): Observable<Article> {
+    return this.http.post(this.url, article).pipe(
+      map(this.http.extractData),
+      catchError(this.http.handleError)
+    )
+  }
+
```
Comment (jpbalarini) [inline src/app/services/articles.service.ts]: remove line

File: src/app/services/authentication.service.ts [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+      catchError(this.http.handleError)
+    );
+  }
+
+  logout() {
+    localStorage.removeItem('id');
+    localStorage.removeItem('email');
+    localStorage.removeItem('token');
+  }
+
```
Comment (jpbalarini) [inline src/app/services/authentication.service.ts]: here

File: src/app/comments/comments.component.ts [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+  comments: Comment[] = [];
+  private url = environment.APIUrl;
+
+  public id: number = 0;
+
+  constructor(private commentsService: CommentsService) { }
+
+  ngOnInit() {
```
Comment (guillermoap) [inline src/app/comments/comments.component.ts]: Remove if not used

File: src/app/login/login.component.css [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+  font-weight: bold;
+  margin: 20px;
+  text-align: center;
+}
+
+form {
+  display: inline-block;
+  margin: 100px auto;
+  padding: 20px 50px;
+  border: 1px solid dodgerblue;
```
Comment (guillermoap) [inline src/app/login/login.component.css]: Use scss variables for colors

File: src/app/login/login.component.css [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+
+input {
+  font-size: 20px;
+  border-radius: 5px;
+  margin: 5px 0;
+  width: 100%;
+}
+
+.ng-valid[required], .ng-valid.required  {
+  border-left: 5px solid #42A948; /* green */
```
Comment (guillermoap) [inline src/app/login/login.component.css]: Same here

File: src/app/login/login.component.css [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+.ng-invalid:not(form)  {
+  border-left: 5px solid #a94442; /* red */
+}
+
+button {
+  margin: 10px;
+  background-color: rgb(17, 188, 201);
```
Comment (guillermoap) [inline src/app/login/login.component.css]: Same here

File: src/app/login/login.component.ts [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+@Component({
+  selector: 'app-login',
+  templateUrl: './login.component.html',
+  styleUrls: ['./login.component.css']
+})
+export class LoginComponent implements OnInit {
+
+  user = new User("default", "123");
+  returnUrl: string;
+  response;
```
Comment (guillermoap) [inline src/app/login/login.component.ts]: Add type

File: src/app/login/login.component.ts [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+  ngOnInit() {
+    // Reset login status
+    this.authenticationService.logout();
+
+    this.returnUrl = this.route.snapshot.queryParams['returnUrl'] || '/';
+  }
+
+  onSubmit() {
+    this.authenticationService.login(this.user).subscribe(
+      data => {
```
Comment (guillermoap) [inline src/app/login/login.component.ts]: If you only have one line don't use the `{}` block

File: src/app/models/user.ts (hunk 9 above)
Comment (guillermoap) [inline src/app/models/user.ts]: Remove space

File: src/app/page-not-found/page-not-found.component.ts [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+@Component({
+  selector: 'app-page-not-found',
+  templateUrl: './page-not-found.component.html',
+  styleUrls: ['./page-not-found.component.css']
+})
+export class PageNotFoundComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit() {
```
Comment (guillermoap) [inline src/app/page-not-found/page-not-found.component.ts]: Remove if not used

File: src/app/services/authentication.service.ts [hunk 20]
```diff
@@ -0,0 +1,40 @@
...
+export class AuthenticationService {
+  private url = `${environment.APIUrl}/users`;
+
+  constructor(private http: HttpService) { }
+
+  login(user: User) {
+    let headers = new Headers({ 'Content-Type': 'application/json' });
+    let options = new RequestOptions({ headers: headers });
+
+    return this.http.post(`${this.url}/sign_in`,  JSON.stringify({'user': {'email': user.email, 'password': user.password}}), options).pipe(
```
Comment (guillermoap) [inline src/app/services/authentication.service.ts]: Line too long

Comment (guillermoap) [reply src/app/services/authentication.service.ts]: Add spaces for `{}`

File: src/app/articles/articles.component.ts (hunk 4 above)
Comment (guillermoap) [reply src/app/articles/articles.component.ts]: I think it's unnecessary to have the `{}` and the `;`

**Pull request merged**